src/get-started/cryptocurrency-advanced.md

```diff
@@ -0,0 +1,801 @@
+[here](https://github.com/exonum/exonum/tree/master/examples/cryptocurrency-advanced).
+
+Besides, unlike its predecessor, the tutorial contains a
+[client part]( https://github.com/exonum/exonum/tree/master/examples/cryptocurrency-advanced/frontend)
+which provides for simple interaction with the service.
+
+## Create Rust Project
+
+To build the service you must have [Node.js](https://nodejs.org/en/download/)
+and the latest version of
+[Rust](https://rustup.rs/) compiler installed on your PC.
+
+Create a crate with the **exonum** crate as a dependency.
+
+```sh
+cargo new cryptocurrency-advanced
+```
+
+Add necessary dependencies to `Cargo.toml` in the project directory:
+
+```
+[dependencies]
+exonum = "0.9.0"
+exonum-configuration = "0.9.0"
+serde = "1.0"
+serde_derive = "1.0"
+failure = "=0.1.1"
+```
+
+Then import said dependencies into `src/lib.rs` file. We will use them to fetch
+the necessary structures, functions, types, traits etc.:
+
+```rust
+#[macro_use]
+extern crate exonum;
+#[macro_use]
+extern crate failure;
+extern crate serde;
+#[macro_use]
+extern crate serde_derive;
+```
+
+For convenience reasons we decided to divide the code into 4 submodules, each
+corresponding to a certain part of the
+service business logic. Let’s announce them below:
+
+```rust
+pub mod api;
+pub mod schema;
+pub mod transactions;
+pub mod wallet;
+```
+
+## Constants
+
+Let’s define the constants to be used further:
+
+```rust
+/// Unique service ID.
+const CRYPTOCURRENCY_SERVICE_ID: u16 = 128;
+/// Name of the service.
+pub const SERVICE_NAME: &str = "cryptocurrency";
+/// Initial balance of the wallet.
+const INITIAL_BALANCE: u64 = 100;
+```
+
+Implement a service structure and realize a trait for it with all the necessary
+methods and credentials:
+
+```rust
+/// Exonum `Service` implementation.
+#[derive(Default, Debug)]
+pub struct CurrencyService;
+
+impl Service for CurrencyService {
+    // Indicate service name
+    fn service_name(&self) -> &str {
+        SERVICE_NAME
+    }
+
+    // Indicate service ID.
+    fn service_id(&self) -> u16 {
+        CRYPTOCURRENCY_SERVICE_ID
+    }
+
+    // Indicate the way the state hash of the database is calculated.
+    fn state_hash(&self, view: &dyn Snapshot) -> Vec<Hash> {
+        let schema = CurrencySchema::new(view);
+        schema.state_hash()
+    }
+
+    // Interface to create transactions from raw data.
+    fn tx_from_raw(&self, raw: RawTransaction) -> Result<Box<dyn Transaction>, EncodingError> {
+        WalletTransactions::tx_from_raw(raw).map(Into::into)
+    }
+
+    // Indicate public API.
+    fn wire_api(&self, builder: &mut ServiceApiBuilder) {
+        api::CryptocurrencyApi::wire(builder);
+    }
+}
+```
+
+Unlike in previous tutorial where node configuration was realized manually, in
+the advanced tutorial we implement a
+trait that allows us to launch the project with the help of `NodeBuilder`.
+`NodeBuilder` is an abstract that allows to
```

Comment:
> `NodeBuilder` is an abstract <_missing noun_>

Maybe... "utility object" or just omit it: "`NodeBuilder` allows to easily configure a node for launch"?

Reply:
abstract is a noun here